coders/jxl.c

```diff
@@ -622,7 +622,7 @@ static MagickBooleanType WriteJXLImage(const ImageInfo *image_info,Image *image,
     }
   memset(&format,0,sizeof(format));
   JXLSetFormat(image,&format);
-  memset(&basic_info,0,sizeof(basic_info));
+  JxlEncoderInitBasicInfo(&basic_info);
```

Comment:
Is this a breaking change in version `0.6`? Could this cause the issue reported in #4319?

Reply:
This is sort of a breaking change, the breaking change is to reject basic_info with invalid values. The issue is that memset to 0 all fields in the basic_info is not a valid basic info because in some fields 0 is not a valid value, which in 0.6 it causes the function to return error. Before 0.6 the invalid value was silently ignored. In addition to that, while 0 is a valid value in other fields, the default is not great (example, intensity_target = 0 is not a good choice), so we now provide a function to set reasonable defaults.

Reply:
Thanks for the info.